Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/preprocessor/transform/DifferenceTransformer.java

```diff
@@ -76,4 +76,12 @@ public double[] transformValues(int internalTimeStamp, double[] inputPoint, doub
         return super.transformValues(internalTimeStamp, input, null, initials, clipFactor);
     }
 
+    @Override
```

Comment:
This is where the transformations are happening. Invert is actually the inverse of this transformValues process. Here we are taking the difference; in invert we are adding previous[I] back.

Reply:
I meant to ask the purpose of getShift, not transformValues. Can ask you offline.

Reply:
Ah, got it. The transformations considered in TRCF all have the following form -- first we shift the data, and then we scale it. (thus inversion is first scale back and then shift). Now we can use invert() in the preprocessor. However for the predictor-corrector, it would break the pre-detect-post abstraction (by having some extra invert()). It therefore makes sense if we add the shift and scale to the transcript for the point in question and we avoid the back and forth function calls into preprocessor. In case of WeightedTransform the shift is 0, but for others, say for difference, shift is over overriden.